bigquery/google/cloud/bigquery/magics.py

```diff
@@ -0,0 +1,251 @@
+
+        %%bigquery [<destination_var>] [--project <project>] [--use_legacy_sql]
+                   [--verbose]
+        <query>
+
+    Parameters:
+
+    * ``<destination_var>`` (optional, line argument):
+        variable to store the query results.
+    * ``--project <project>`` (optional, line argument):
+        Project to use for running the query. Defaults to the client's default
+        project.
+    * ``--use_legacy_sql`` (optional, line argument):
+        Runs the query using Legacy SQL syntax. Defaults to Standard SQL if
+        this argument not used.
+    * ``--verbose`` (optional, line argument):
+        If this flag is used, information including the query job ID and the
+        amount of time for the query to complete will not be cleared after the
+        query is finished. By default, this information will be displayed but
+        will be cleared after the query is finished.
+    * ``<query>`` (required, cell argument):
+        SQL query to run.
+
+    Returns:
+        A :class:`pandas.DataFrame` with the query results.
+
+    .. note::
+        All queries run using this magic will run using the context client
+        (:func:`Context.client <google.cloud.bigquery.magics.Context.client>`)
+
+    Examples:
+        The following examples can be run in an IPython notebook after loading
+        the bigquery IPython extension (see ``In[1]``) and setting up
+        Application Default Credentials.
+
+    .. code-block:: python
+
+        In [1]: %load_ext google.cloud.bigquery
+
+        In [2]: %%bigquery
+           ...: SELECT name, SUM(number) as count
+           ...: FROM `bigquery-public-data.usa_names.usa_1910_current`
+           ...: GROUP BY name
+           ...: ORDER BY count DESC
+           ...: LIMIT 3
+
+        Out[2]:       name    count
+           ...: -------------------
+           ...: 0    James  4987296
+           ...: 1     John  4866302
+           ...: 2   Robert  4738204
+
+        In [3]: %%bigquery df --project my-alternate-project --verbose
+           ...: SELECT name, SUM(number) as count
+           ...: FROM `bigquery-public-data.usa_names.usa_1910_current`
+           ...: WHERE gender = 'F'
+           ...: GROUP BY name
+           ...: ORDER BY count DESC
+           ...: LIMIT 3
+        Executing query with job ID: bf633912-af2c-4780-b568-5d868058632b
+        Query executing: 2.61s
+        Query complete after 2.92s
+
+        Out[3]:          name    count
+           ...: ----------------------
+           ...: 0        Mary  3736239
+           ...: 1    Patricia  1568495
+           ...: 2   Elizabeth  1519946
+
+        In [4]: df
+
+        Out[4]:          name    count
+           ...: ----------------------
+           ...: 0        Mary  3736239
+           ...: 1    Patricia  1568495
+           ...: 2   Elizabeth  1519946
+
+"""
+
+from __future__ import print_function
+
+import time
+from concurrent import futures
+
+try:
+    from IPython import get_ipython
```

Comment:
I'm not seeing this in the IPython docs. http://ipython.readthedocs.io/en/stable/api/generated/IPython.html

I do see: http://ipython.readthedocs.io/en/stable/api/generated/IPython.core.getipython.html

Should we be more specific about what versions of IPython we support?

Reply:
I think it works because of [this line](https://github.com/ipython/ipython/blob/48eab58c7a96a66b67b15b926a6143db86d16a69/IPython/__init__.py#L52) in the IPython init file. I generally see it being used in this way in other code rather than `IPython.core.getipython.get_ipython()`, but I can change it if it's confusing.